### PR TITLE
Remove queries from query folder and add tag dimension common dimensions across all queries

### PR DESCRIPTION
**Pull request by Priyanka-Chatterjee-2000**
### Checklist
- [ ] Issue(s) linked

### CODE REVIEW

File: query/appservice.sp [hunk 1]
```diff
@@ -0,0 +1,681 @@
...
+        when (arguments -> 'site_config' ->> 'ftps_state')::text = 'FtpsOnly' then ' FTPS enabled'
+        else '  FTPS disabled'
+      end || '.' reason
+      ${local.tag_dimensions_sql}
+      ${local.common_dimensions_sql}
+    from
+      terraform_resource
+    where
+      type = 'azurerm_function_app'
+    union
```
Comment (madhushreeray30) [inline query/appservice.sp:347]:
```suggestion

    union
```

File: query/compute.sp [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+        else 'ok'
+      end as status,
+      name || case
+        when d.vm_name is null then ' have guest configuration extension not installed'
+        else ' have guest configuration extension installed'
+      end || '.' reason
+      ${local.tag_dimensions_sql}
+      ${local.common_dimensions_sql}
+    from
+      all_windows_vm as c left join vm_guest_configuration as d on c.name = d.vm_name;
```
Comment (madhushreeray30) [inline query/compute.sp]:
```suggestion
      all_windows_vm as c 
      left join vm_guest_configuration as d on c.name = d.vm_name;
```

File: query/cognitiveservices.sp [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        else 'alarm'
+      end as status,
+      c.name || case
+        when (b.arguments ->> 'cognitive_account_id') is not null then ' encrypted with CMK'
+        else ' not encrypted with CMK'
+      end || '.' reason
+      ${replace(local.tag_dimensions_qualifier_sql, "__QUALIFIER__", "c.")}
+      ${replace(local.common_dimensions_qualifier_sql, "__QUALIFIER__", "c.")}
+    from
+      all_cognitive_account as c left join cognitive_account_cmk as b on c.name = (split_part((b.arguments ->> 'cognitive_account_id'), '.', 2))
```
Comment (madhushreeray30) [inline query/cognitiveservices.sp]:
```suggestion
      all_cognitive_account as c 
      left join cognitive_account_cmk as b on c.name = (split_part((b.arguments ->> 'cognitive_account_id'), '.', 2))
```

File: query/compute.sp [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+        from
+          terraform_resource
+        where
+          type = 'azurerm_virtual_machine_extension'
+    ),
+    vm_guest_configuration as (
+      select
+        split_part((b.arguments ->> 'virtual_machine_id'), '.', 2) as vm_name
+      from
+        all_windows_vm as a left join vm_extensions as b on  (split_part((b.arguments ->> 'virtual_machine_id'), '.', 2)) = a.name
```
Comment (madhushreeray30) [inline query/compute.sp]:
```suggestion
        all_windows_vm as a 
        left join vm_extensions as b on  (split_part((b.arguments ->> 'virtual_machine_id'), '.', 2)) = a.name
```

File: query/compute.sp [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+        from
+          terraform_resource
+        where
+          type = 'azurerm_virtual_machine_extension'
+    ),
+    vm_guest_configuration as (
+      select
+        split_part((b.arguments ->> 'virtual_machine_id'), '.', 2) as vm_name
+      from
+        all_vm as a left join vm_extensions as b on (split_part((b.arguments ->> 'virtual_machine_id'), '.', 2)) = a.name
```
Comment (madhushreeray30) [inline query/compute.sp]:
```suggestion
        all_vm as a 
        left join vm_extensions as b on (split_part((b.arguments ->> 'virtual_machine_id'), '.', 2)) = a.name
```

File: query/compute.sp [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+        else 'ok'
+      end as status,
+      name || case
+        when d.vm_name is null then ' have guest configuration extension not installed'
+        else ' have guest configuration extension installed'
+      end || '.' reason
+      ${local.tag_dimensions_sql}
+      ${local.common_dimensions_sql}
+    from
+      all_vm as c left join vm_guest_configuration as d on c.name = d.vm_name;
```
Comment (madhushreeray30) [inline query/compute.sp]:
```suggestion
      all_vm as c 
      left join vm_guest_configuration as d on c.name = d.vm_name;
```

File: query/compute.sp [hunk 7]
```diff
@@ -0,0 +1,284 @@
...
+        from
+          terraform_resource
+        where
+          type = 'azurerm_virtual_machine_extension'
+    ),
+    vm_guest_configuration as (
+      select
+        split_part((b.arguments ->> 'virtual_machine_id'), '.', 2) as vm_name
+      from
+        all_linux_vm as a left join vm_extensions as b on  split_part((b.arguments ->> 'virtual_machine_id'), '.', 2) = a.name
```
Comment (madhushreeray30) [inline query/compute.sp]:
```suggestion
        all_linux_vm as a 
        left join vm_extensions as b on  split_part((b.arguments ->> 'virtual_machine_id'), '.', 2) = a.name
```

File: query/compute.sp [hunk 8]
```diff
@@ -0,0 +1,284 @@
...
+        else 'ok'
+      end as status,
+      name || case
+        when d.vm_name is null then ' have guest configuration extension not installed'
+        else ' have guest configuration extension installed'
+      end || '.' reason
+      ${local.tag_dimensions_sql}
+      ${local.common_dimensions_sql}
+    from
+      all_linux_vm as c left join vm_guest_configuration as d on c.name = d.vm_name;
```
Comment (madhushreeray30) [inline query/compute.sp]:
```suggestion
      all_linux_vm as c 
      left join vm_guest_configuration as d on c.name = d.vm_name;
```

File: query/compute.sp [hunk 9]
```diff
@@ -0,0 +1,284 @@
...
+  EOQ
+}
+
+query "compute_vm_and_scale_set_encryption_at_host_enabled" {
+  sql = <<-EOQ
+    (select
```
Comment (madhushreeray30) [inline query/compute.sp]:
```suggestion
    (
    select
```

File: query/compute.sp [hunk 10]
```diff
@@ -0,0 +1,284 @@
...
+      ${local.tag_dimensions_sql}
+      ${local.common_dimensions_sql}
+    from
+      terraform_resource
+    where
+      type = 'azurerm_windows_virtual_machine' or
+      type = 'azurerm_linux_virtual_machine'
+    )
+  EOQ
+}
```
Comment (madhushreeray30) [inline query/compute.sp:225]: this gives error > `each UNION query must have the same number of columns (SQLSTATE 42601)`

File: query/compute.sp [hunk 11]
```diff
@@ -0,0 +1,284 @@
...
+        from
+          terraform_resource
+        where
+          type = 'azurerm_virtual_machine_extension'
+    ),
+    vm_amtimalware_extension as (
+      select
+        split_part((b.arguments ->> 'virtual_machine_id'), '.', 2) as vm_name
+      from
+        all_vm as a left join vm_extensions as b on (split_part((b.arguments ->> 'virtual_machine_id'), '.', 2)) = a.name
```
Comment (madhushreeray30) [inline query/compute.sp]:
```suggestion
        all_vm as a 
        left join vm_extensions as b on (split_part((b.arguments ->> 'virtual_machine_id'), '.', 2)) = a.name
```

File: query/compute.sp [hunk 12]
```diff
@@ -0,0 +1,284 @@
...
+      end as status,
+      name || case
+        when d.vm_name is null then ' IaaSAntimalware extension not installed'
+        else ' IaaSAntimalware extension installed'
+      end || '.' reason,
+      path || ':' || start_line
+      ${local.tag_dimensions_sql}
+      ${local.common_dimensions_sql}
+    from
+      all_vm as c left join vm_amtimalware_extension as d on c.name = d.vm_name;
```
Comment (madhushreeray30) [inline query/compute.sp:288]:
```suggestion
      all_vm as c 
      left join vm_amtimalware_extension as d on c.name = d.vm_name;
```

File: query/keyvault.sp [hunk 13]
```diff
@@ -0,0 +1,258 @@
...
+        *
+      from
+        terraform_resource
+      where
+        type = 'azurerm_monitor_diagnostic_setting' and (arguments ->> 'target_resource_id') like '%azurerm_key_vault_managed_hardware_security_module.%'
+    ), hsm_key_vaults_logging as (
+      select
+        kv.name as kv_name
+      from
+        hsm_key_vaults as kv left join  diagnostic_setting as ds on kv.name = (split_part((ds.arguments ->> 'target_resource_id'), '.', 2))
```
Comment (madhushreeray30) [inline query/keyvault.sp]:
```suggestion
        hsm_key_vaults as kv 
        left join  diagnostic_setting as ds on kv.name = (split_part((ds.arguments ->> 'target_resource_id'), '.', 2))
```

File: query/keyvault.sp [hunk 14]
```diff
@@ -0,0 +1,258 @@
...
+      from
+        terraform_resource
+      where
+        type = 'azurerm_monitor_diagnostic_setting' and (arguments ->> 'target_resource_id') like '%azurerm_key_vault.%'
+    ), key_vaults_logging as (
+      select
+        kv.name as kv_name,
+        ds.arguments
+      from
+        key_vaults as kv left join diagnostic_setting as ds on kv.name = (split_part((ds.arguments ->> 'target_resource_id'), '.', 2))
```
Comment (madhushreeray30) [inline query/keyvault.sp]:
```suggestion
        key_vaults as kv 
        left join diagnostic_setting as ds on kv.name = (split_part((ds.arguments ->> 'target_resource_id'), '.', 2))
```

File: query/monitor.sp [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+        when (arguments ->> 'container_access_type') ilike 'Private' then ' container insights-operational-logs storing activity logs not publicly accessible'
+        else ' container insights-operational-logs storing activity logs publicly accessible'
+      end || '.' reason,
+      path || ':' || start_line
+      ${local.tag_dimensions_sql}
+      ${local.common_dimensions_sql}
+    from
+      terraform_resource
+    where
+      type = 'azurerm_storage_container' and (arguments ->> 'name') ilike 'insights-operational-logs';
```
Comment (madhushreeray30) [inline query/monitor.sp]:
```suggestion
      type = 'azurerm_storage_container' 
      and (arguments ->> 'name') ilike 'insights-operational-logs';
```

File: query/mysql.sp [hunk 16]
```diff
@@ -0,0 +1,121 @@
...
+      a.type || ' ' || a.name as resource,
+      case
+        when (s.arguments ->> 'server_id') is not null then 'ok'
+        else 'alarm'
+      end as status,
+      a.name || case
+        when (s.arguments ->> 'server_id') is not null then ' encrypted with CMK'
+        else ' not encrypted with CMK'
+      end || '.' reason,
+      a.path || ':' || a.start_line
```
Comment (madhushreeray30) [inline query/mysql.sp]:
```suggestion
```

File: query/mysql.sp [hunk 17]
```diff
@@ -0,0 +1,121 @@
...
+    select
+      a.type || ' ' || a.name as resource,
+      case
+        when (s.arguments ->> 'server_id') is not null then 'ok'
+        else 'alarm'
+      end as status,
+      a.name || case
+        when (s.arguments ->> 'server_id') is not null then ' encrypted with CMK'
+        else ' not encrypted with CMK'
+      end || '.' reason,
```
Comment (madhushreeray30) [inline query/mysql.sp]:
```suggestion
      end || '.' reason
```

File: query/network.sp [hunk 18]
```diff
@@ -0,0 +1,113 @@
...
+    select
+      a.type || ' ' || a.name as resource,
+      case
+        when (s.arguments ->> 'subnet_id') is not null then 'ok'
+        else 'alarm'
+      end as status,
+      a.name || case
+        when (s.arguments ->> 'subnet_id') is not null then ' associated with subnet'
+        else ' not associated with subnet'
+      end || '.' reason,
```
Comment (madhushreeray30) [inline query/network.sp]:
```suggestion
      end || '.' reason
```

File: query/network.sp [hunk 19]
```diff
@@ -0,0 +1,113 @@
...
+      a.path || ':' || a.start_line
```
Comment (madhushreeray30) [inline query/network.sp]:
```suggestion
```

File: query/springcloud.sp [hunk 20]
```diff
@@ -0,0 +1,22 @@
...
+    select
+      type || ' ' || name as resource,
+      case
+        when (arguments -> 'network') is null then 'alarm'
+        else 'ok'
+      end status,
+      name || case
+        when (arguments -> 'network') is null then ' network injection disabled'
+        else ' network injection enabled'
+      end || '.' reason,
```
Comment (madhushreeray30) [inline query/springcloud.sp]:
```suggestion
      end || '.' reason
```

File: query/springcloud.sp [hunk 21]
```diff
@@ -0,0 +1,22 @@
...
+      path || ':' || start_line
```
Comment (madhushreeray30) [inline query/springcloud.sp]:
```suggestion
```

File: query/sql.sp [hunk 22]
```diff
@@ -0,0 +1,256 @@
...
+    select
+      a.type || ' ' || a.name as resource,
+      case
+        when (s.arguments ->> 'server_id') is not null then 'ok'
+        else 'alarm'
+      end as status,
+      a.name || case
+        when (s.arguments ->> 'server_name') is not null then ' audit retention greater than 90 days'
+        else ' audit retention less than 90 days'
+      end || '.' reason,
```
Comment (madhushreeray30) [inline query/sql.sp]:
```suggestion
      end || '.' reason
```

File: query/sql.sp [hunk 23]
```diff
@@ -0,0 +1,256 @@
...
+      a.path || ':' || a.start_line
```
Comment (madhushreeray30) [inline query/sql.sp]:
```suggestion
```

File: steampipe.spvars.example [hunk 24]
```diff
@@ -0,0 +1,4 @@
+# Dimensions
```
Comment (madhushreeray30) [inline steampipe.spvars.example:1]:
```suggestion
# Dimensions
# Possible values for common dimensions: "path" and "connection_name"
```

File: steampipe.spvars.example [hunk 25]
```diff
@@ -0,0 +1,4 @@
...
+common_dimensions = [ "path" ]
```
Comment (madhushreeray30) [inline steampipe.spvars.example]:
```suggestion
common_dimensions = ["path"]
```

File: query/monitor.sp [hunk 26]
```diff
@@ -0,0 +1,66 @@
...
+      case
+        when (arguments -> 'container_access_type') is null then 'ok'
+        when (arguments ->> 'container_access_type') ilike 'Private' then 'ok'
+        else 'alarm'
+      end status,
+      name || case
+        when (arguments -> 'container_access_type') is null then ' container insights-operational-logs storing activity logs not publicly accessible'
+        when (arguments ->> 'container_access_type') ilike 'Private' then ' container insights-operational-logs storing activity logs not publicly accessible'
+        else ' container insights-operational-logs storing activity logs publicly accessible'
+      end || '.' reason,
```
Comment (madhushreeray30) [inline query/monitor.sp]:
```suggestion
      end || '.' reason
```

File: query/monitor.sp [hunk 27]
```diff
@@ -0,0 +1,66 @@
...
+      path || ':' || start_line
```
Comment (madhushreeray30) [inline query/monitor.sp]:
```suggestion
```

File: query/compute.sp [hunk 28]
```diff
@@ -0,0 +1,284 @@
...
+    select
+      type || ' ' || name as resource,
+      case
+        when d.vm_name is null then 'alarm'
+        else 'ok'
+      end as status,
+      name || case
+        when d.vm_name is null then ' IaaSAntimalware extension not installed'
+        else ' IaaSAntimalware extension installed'
+      end || '.' reason,
```
Comment (madhushreeray30) [inline query/compute.sp]:
```suggestion
      end || '.' reason
```

File: query/compute.sp [hunk 29]
```diff
@@ -0,0 +1,284 @@
...
+      path || ':' || start_line
```
Comment (madhushreeray30) [inline query/compute.sp]:
```suggestion
```

File: query/compute.sp [hunk 30]
```diff
@@ -0,0 +1,284 @@
...
+    select
+      type || ' ' || name as resource,
+      case
+        when (arguments ->> 'resource_group_name') is not null then 'ok'
+        else 'alarm'
+      end status,
+      name || case
+        when (arguments ->> 'resource_group_name') is not null then ' uses azure resource manager'
+        else ' uses azure resource manager'
+      end || '.' reason,
```
Comment (madhushreeray30) [inline query/compute.sp]:
```suggestion
      end || '.' reason
```

File: query/compute.sp [hunk 31]
```diff
@@ -0,0 +1,284 @@
...
+      path || ':' || start_line
```
Comment (madhushreeray30) [inline query/compute.sp]:
```suggestion
```

File: query/compute.sp [hunk 32]
```diff
@@ -0,0 +1,284 @@
...
+      case
+        when (arguments -> 'encryption_at_host_enabled') is null then 'alarm'
+        when (arguments -> 'encryption_at_host_enabled')::boolean then 'ok'
+        else 'alarm'
+      end status,
+      name || case
+        when (arguments -> 'encryption_at_host_enabled') is null then ' ''encryption_at_host_enabled'' is not defined'
+        when (arguments -> 'encryption_at_host_enabled')::boolean then ' encryption at host enabled'
+        else ' encryption at host disabled'
+      end || '.' reason,
```
Comment (madhushreeray30) [inline query/compute.sp]:
```suggestion
      end || '.' reason
```

File: query/compute.sp [hunk 33]
```diff
@@ -0,0 +1,284 @@
...
+      path || ':' || start_line
```
Comment (madhushreeray30) [inline query/compute.sp]:
```suggestion
```

File: query/batch.sp [hunk 34]
```diff
@@ -0,0 +1,65 @@
...
+        *
+      from
+        terraform_resource
+      where
+        type = 'azurerm_monitor_diagnostic_setting' and (arguments ->> 'target_resource_id') like '%azurerm_batch_account.%'
+    ), batch_account_logging as (
+      select
+        ba.name as ba_name
+      from
+        batch_accounts as ba left join diagnostic_setting as ds on ba.name = (split_part((ds.arguments ->> 'target_resource_id'), '.', 2))
```
Comment (madhushreeray30) [inline query/batch.sp]:
```suggestion
        batch_accounts as ba 
        left join diagnostic_setting as ds on ba.name = (split_part((ds.arguments ->> 'target_resource_id'), '.', 2))
```

File: query/compute.sp [hunk 35]
```diff
@@ -0,0 +1,284 @@
...
+
+query "compute_vm_guest_configuration_installed_windows" {
+  sql = <<-EOQ
+    with all_windows_vm as (
+      select
+      *
+      from
+        terraform_resource
+      where
+        type = 'azurerm_virtual_machine' and (arguments -> 'os_profile_windows_config') is not null
```
Comment (madhushreeray30) [inline query/compute.sp]:
```suggestion
        type = 'azurerm_virtual_machine' 
        and (arguments -> 'os_profile_windows_config') is not null
```

File: query/compute.sp [hunk 36]
```diff
@@ -0,0 +1,284 @@
...
+
+query "compute_vm_guest_configuration_installed_linux" {
+  sql = <<-EOQ
+    with all_linux_vm as (
+      select
+      *
+      from
+        terraform_resource
+      where
+        type = 'azurerm_virtual_machine' and (arguments -> 'os_profile_linux_config') is not null
```
Comment (madhushreeray30) [inline query/compute.sp]:
```suggestion
        type = 'azurerm_virtual_machine' 
        and (arguments -> 'os_profile_linux_config') is not null
```

File: query/keyvault.sp [hunk 37]
```diff
@@ -0,0 +1,258 @@
...
+        terraform_resource
+      where
+        type = 'azurerm_key_vault_managed_hardware_security_module'
+    ), diagnostic_setting as (
+      select
+        *
+      from
+        terraform_resource
+      where
+        type = 'azurerm_monitor_diagnostic_setting' and (arguments ->> 'target_resource_id') like '%azurerm_key_vault_managed_hardware_security_module.%'
```
Comment (madhushreeray30) [inline query/keyvault.sp]:
```suggestion
        type = 'azurerm_monitor_diagnostic_setting' 
        and (arguments ->> 'target_resource_id') like '%azurerm_key_vault_managed_hardware_security_module.%'
```

File: query/keyvault.sp [hunk 38]
```diff
@@ -0,0 +1,258 @@
...
+        terraform_resource
+      where
+        type = 'azurerm_key_vault'
+    ), diagnostic_setting as (
+      select
+        *
+      from
+        terraform_resource
+      where
+        type = 'azurerm_monitor_diagnostic_setting' and (arguments ->> 'target_resource_id') like '%azurerm_key_vault.%'
```
Comment (madhushreeray30) [inline query/keyvault.sp]:
```suggestion
        type = 'azurerm_monitor_diagnostic_setting' 
        and (arguments ->> 'target_resource_id') like '%azurerm_key_vault.%'
```

File: query/storage.sp [hunk 39]
```diff
@@ -0,0 +1,372 @@
...
+        select
+          name,
+          type,
+          path,
+          start_line,
+          split_part((arguments ->> 'storage_account_name'), '.',2) as storage_account_name
+        from
+          terraform_resource
+        where
+          type = 'azurerm_storage_account_network_rules' and (arguments ->> 'default_action') = 'Deny'
```
Comment (madhushreeray30) [inline query/storage.sp]:
```suggestion
          type = 'azurerm_storage_account_network_rules' 
          and (arguments ->> 'default_action') = 'Deny'
```

File: query/storage.sp [hunk 40]
```diff
@@ -0,0 +1,372 @@
...
+        select
+          name,
+          type,
+          path,
+          start_line,
+          split_part((arguments ->> 'storage_account_name'), '.',2) as storage_account_name
+        from
+          terraform_resource
+        where
+          type = 'azurerm_storage_account_network_rules' and (arguments ->> 'bypass') like '%AzureServices%'
```
Comment (madhushreeray30) [inline query/storage.sp]:
```suggestion
          type = 'azurerm_storage_account_network_rules' 
          and (arguments ->> 'bypass') like '%AzureServices%'
```

**Review by madhushreeray30 — COMMENTED** [review]
@Priyanka-Chatterjee-2000 please have a look at the comments. Thanks!

**Comment by misraved** [comment]
@madhushreeray30 @khushboo9024 could you please update the doc files to reflect the configuration and usage of common and tag dimensions?

**Review by misraved — APPROVED** [review]

**Pull request merged**